### ISSUE TITLE
Assertion `!(IsVariadic && IsIndirect)' failed: llvm::SDDbgValue::SDDbgValue(llvm::BumpPtrAllocator &, llvm::DIVariable *, llvm::DIExpression *, ArrayRef< llvm::SDDbgOperand>, ArrayRef<llvm::SDNode *>, bool, llvm::DebugLoc, unsigned int, bool)

### ISSUE THREAD

**Issue by dcci** [llvm/llvm-project#72447]
To repro:
```
./llc -O3 reduced.ll.txt
```
(file attached)
[reduced.ll.txt](https://github.com/llvm/llvm-project/files/13370372/reduced.ll.txt)

Points to:
```
commit 3ab03ad37d48d60596fc7f50f6c76794030baaf3
Author: David <david.stenberg@ericsson.com>
Date:   Wed Oct 25 12:51:49 2023 +0200

    [SelectionDAG] Salvage debug info for non-constant ADDs (#68981)
    
    Teach SelectionDAG::salvageDebugInfo() to salvage debug information for
    ADD nodes where the RHS is non-constant.
    
    Co-authored-by: Mikael Holmen <mikael.holmen@ericsson.com>
    
    - [DebugInfo] Precommit testcase for pointer addition with unknown
    offset
    - [SelectionDAG] Salvage debug info for non-constant ADDs
    
    ---------
    
    Co-authored-by: Mikael Holmen <mikael.holmen@ericsson.com>

 llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp     | 43 ++++++++++++----
 .../pointer-add-unknown-offset-debug-info.mir      | 60 ++++++++++++++++++++++
 2 files changed, 93 insertions(+), 10 deletions(-)
```

**Comment by dcci** [comment]
@dstenb can you please help taking a look?

**Comment by dstenb** [comment]
Fixed by #72645.
